### PR TITLE
New API examples porting

### PR DESCRIPTION
**Pull request by AnastaZIuk**
we need to resolve include errors first

### CODE REVIEW

File: examples_tests/common/CommonAPI.h [hunk 1]
```diff
@@ -25,23 +27,36 @@ class CommonAPI
...
+			EQT_COUNT
+		};
+
 		nbl::core::smart_refctd_ptr<CWindowT> window;
 		nbl::core::smart_refctd_ptr<nbl::video::IAPIConnection> apiConnection;
 		nbl::core::smart_refctd_ptr<nbl::video::ISurface> surface;
 		nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice> logicalDevice;
 		nbl::core::smart_refctd_ptr<nbl::video::IPhysicalDevice> physicalDevice;
-		nbl::video::IGPUQueue* queue;
+		std::array<nbl::video::IGPUQueue*, EQT_COUNT> queues;
```
Comment (AnastaZIuk) [inline examples_tests/common/CommonAPI.h:48]: @Erfan-Ahmadi see new changes

**Comment by AnastaZIuk** [comment]
soon @sadiuk will be testing loaders and writers after applied changes, then we will be able to use them with no doubt

File: examples_tests/common/CommonAPI.h [hunk 2]
```diff
@@ -1,29 +1,97 @@
...
+
+	static nbl::core::smart_refctd_ptr<nbl::system::ISystem> createSystem()
 	{
 		using namespace nbl;
 		using namespace core;
 		using namespace system;
 		smart_refctd_ptr<ISystemCaller> caller = nullptr;
 #ifdef _NBL_PLATFORM_WINDOWS_
-		caller = make_smart_refctd_ptr<CSystemCallerWin32>();
+		//caller = make_smart_refctd_ptr<nbl::system::CSystemCallerWin32>(); WUT? NOT DEFINED XD
```
Comment (devshgraphicsprogramming) [inline examples_tests/common/CommonAPI.h:94]: missing include?

File: examples_tests/common/CommonAPI.h [hunk 3]
```diff
@@ -1,29 +1,97 @@
...
 #include <nabla.h>
-#include "nbl/system/CSystemWin32.h"
 #if defined(_NBL_PLATFORM_WINDOWS_)
-#include "nbl/system/CColoredStdoutLoggerWin32.h"
-#include <nbl/ui/CWindowWin32.h>
+#include <nbl/ui/CWindowManagerWin32.h>
+#include <nbl/system/ISystem.h>
+#include <nbl/system/CStdoutLogger.h>
+#include <nbl/system/CFileLogger.h>
+#include <nbl/system/CColoredStdoutLoggerWin32.h>
```
Comment (devshgraphicsprogramming) [inline examples_tests/common/CommonAPI.h:8]: they should all be in the relevant `nbl/namespace/declarations.h` and `nbl/namespace/definitions.h` which are included by `nabla.h.in`

File: examples_tests/common/CommonAPI.h (hunk 2 above)
Comment (AnastaZIuk) [reply examples_tests/common/CommonAPI.h]: missed, resolved

File: examples_tests/common/CommonAPI.h [hunk 4]
```diff
@@ -108,7 +105,7 @@ class CommonAPI
 			EQT_COUNT
 		};
 
-		nbl::core::smart_refctd_ptr<CWindowT> window;
+		nbl::core::smart_refctd_ptr<nbl::ui::IWindow> window;
 		nbl::core::smart_refctd_ptr<nbl::video::IAPIConnection> apiConnection;
 		nbl::core::smart_refctd_ptr<nbl::video::ISurface> surface;
 		nbl::core::smart_refctd_ptr<nbl::video::ILogicalDevice> logicalDevice;
```
Comment (devshgraphicsprogramming) [inline examples_tests/common/CommonAPI.h]: just slam a `using nbl::ui` etc. in the class scope

**Comment by AnastaZIuk** [comment]
new troubles:

- no window created
- OBJ to check
- no archive loader
- no CQuantNormalCache read/write functions
- Nabla.h to correct

**Comment by AnastaZIuk** [comment]
@sadiuk I have a list for you that I and either @Erfan-Ahmadi need

we need:

- shaders loading
- loaders (.OBJ and .PNG, .JPG are priorities)
- writers (.PNG is the priority)

to be tested, please use 06 example as your reference (when @Erfan-Ahmadi adjusts to queues comment we can also look at 17. example)

**Comment by AnastaZIuk** [comment]
@AnastaZIuk and @Erfan-Ahmadi ' TODO: 

- [ ] rudimentary camera implementation
- [ ] Sponza examples (all of them)
- [ ] Color Space example
- [ ] Bullet example
- [ ] Any related example
- [ ] ScreenShot extension
- [ ] `CQuantNormalCache` reading/writing
- [ ] Archive loader (@Devsh-Graphics-Programming/seniors should we take care of it?)

@sadiuk 's TODO https://github.com/Devsh-Graphics-Programming/Nabla/pull/166#issuecomment-890993170

- [ ] shaders loading
- [ ] loaders (according to the list priority)
- [ ] writers (according to the list priority)

**Comment by AnastaZIuk** [comment]
@devshgraphicsprogramming https://github.com/Devsh-Graphics-Programming/Nabla/pull/159#discussion_r680962155

**Pull request merged**